File: autoram_addons/autoram_addons/report/technician_commission/technician_commission.js
// Copyright (c) 2016, Ebkar Technology. and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Technician Commission"] = {
	"filters": [
		{
			fieldname: "technician",
			label: "Technician",
			fieldtype: "Link",
			reqd: 1,
			options: "Employee",
			get_query: function () {
				return {
					"filters": {
						"technician_gets_a_commission": 1
					}
				}
			}
		},
		{
			fieldname:"from_date",
			reqd: 1,
			label: __("From Date"),
			fieldtype: "Date",
			default: frappe.datetime.add_months(frappe.datetime.get_today(), -1),
		},
		{
			fieldname:"to_date",
			reqd: 1,
			default: frappe.datetime.get_today(),
			label: __("To Date"),
			fieldtype: "Date",
		},
		{
			fieldname:"sales_invoice",
			label: __("Sales Invoice"),
			fieldtype: "Link",
			options: "Sales Invoice",
			get_query: function () {
				return {
					filters: {
						"docstatus": "1"
					}
				}
			}
		},
		{
			fieldname:"item",
			label: __("Item"),
			fieldtype: "Link",
			options: "Item"
		},
		{
			fieldname:"item_group",
			label: __("Item Group"),
			fieldtype: "Link",
			options: "Item Group"
		},
		{
			fieldname:"currency",
			default: frappe.defaults.get_default("currency"),
			label: __("Currency"),
			fieldtype: "Link",
			options: "Currency"
		},
	],
	after_datatable_render: function(datatable_obj) {
		rows = datatable_obj.getRows();
		if(rows && rows.length > 0 ) {
			last_row = rows[rows.length -1]
			console.log(last_row, datatable_obj)
			last_row[1].content = "<b>Total</b>";
			last_row[9].content = "<b>" + last_row[9].content + "</b>";
			last_row[11].content = "<b>" + last_row[11].content + "</b>";
			datatable_obj.refreshRow(last_row);
		}
	},
};

File: autoram_addons/autoram_addons/report/technician_commission/technician_commission.py
# Copyright (c) 2013, Ebkar Technology. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
from frappe import _
import frappe

def execute(filters=None):
	columns = get_columns(filters)
	data = get_data(filters)


	return columns, data

def get_columns(filters):
	return [
		{
			"fieldname": "technician_name",
			"label": _("Technician Name"),
			"fieldtype": "Link",
			"options": "Employee",
			"width": "176"
		},
		{
			"fieldname": "invoice_name",
			"label": _("Invoice"),
			"fieldtype": "Link",
			"options": "Sales Invoice",
			"width": "144"
		},
		{
			"fieldname": "invoice_currency",
			"label": _("Currency"),
			"fieldtype": "Link",
			"options": "Currency",
			"width": "64"
		},
		{
			"fieldname": "item_name",
			"label": _("Item"),
			"fieldtype": "Link",
			"options": "Item",
			"width": "180"
		},
		{
			"fieldname": "item_uom",
			"label": _("UOM"),
			"fieldtype": "Link",
			"options": "UOM",
			"width": "50"
		},
				{
			"fieldname": "item_qty",
			"label": _("Quantity"),
			"fieldtype": "Data",
			"width": "73"
		},
		{
			"fieldname": "amount",
			"label": _("Amount"),
			"fieldtype": "Currency",
			"width": "150"
		},
		{
			"fieldname": "commission",
			"label": _("Commission Amount"),
			"fieldtype": "Currency",
			"width": "135"
		},
		{
			"fieldname": "commission_rate",
			"label": _("Commission Rate"),
			"fieldtype": "Currency",
			"width": "120"
		}
	]

def get_data(filters):
	tmp_row = {}
	idx = 0
	filters_dict = {}
	data = []
	si_orm = None

	if(filters.get("from_date")):
		filters_dict["posting_date"] = [">=", filters.get("from_date"), "<=", filters.get("to_date")]

	if(filters.get("currency")):
		filters_dict["currency"] = filters.get("currency")

	si_list = frappe.get_all("Sales Invoice", fields=["name", "currency"], filters=filters_dict)
	
	while(idx < len(si_list)):
		# get si orm for each si list item
		si_orm = frappe.get_doc("Sales Invoice", si_list[idx].name)

		for item in si_orm.items:
			if item.technician != filters.technician:
				continue
			tmp_row = {
				"technician_name": item.technician_name,
				"invoice_name": si_orm.name,
				"invoice_currency": si_list[idx].currency,
				"item_name": item.item_name,
				"item_uom": item.uom,
				"item_qty": frappe.utils.flt(item.qty),
				"amount": item.amount,
				"commission": frappe.format_value(item.technician_commission_amount, "Currency"),
				"commission_rate": frappe.format_value(item.technician_commission_rate, "Currency")
			}
			data.append(tmp_row)
			
		idx+=1

	return data